### ISSUE TITLE
Error on table update using a mapped field in where clause

### ISSUE THREAD

**Issue by icomad** [prisma/prisma#2329]
## Bug description

Hi, I'm on 2.0.0-alpha.1174 and I get an error when trying to update a table using a mapped field in the where clause

## How to reproduce

The relevant portion of my prisma schema:

```
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model FriendRequest {
  createdAt        DateTime   @default(now()) @map(name: "created_at")
  toUserId         Int        @map(name: "to_user_id")
  status           ReqStatus  @default(pending)
  fromUserId       Int        @map(name: "from_user_id")
  toUser           User       @relation("friend_requests_to", fields: [toUserId], references: [id])
  fromUser         User       @relation("friend_requests_from", fields: [fromUserId], references: [id])

  @@id([fromUserId, toUserId])
  @@map(name: "friend_requests")
}
```

The code to update the table:
```
await prisma.friendRequest.update({
    where: {
        fromUserId_toUserId: {fromUserId: 1, toUserId: 2}
    }, 
    data: {status: "accepted"}
});
```
Note: the hardcoded `fromUserId: 1` and `toUserId: 2` are present in the database

The above code does not work and throw this error:
```
UnhandledPromiseRejectionWarning: Error:
Invalid `prisma.friendRequest.update()` invocation in /path/to/the/code
``` 

followed by this one:
```
Error in Prisma Client:
PANIC: called `Option::unwrap()` on a `None` value in
query-engine/core/src/query_graph_builder/extractors/filters.rs:279:35
```

If i change the prisma schema to this:
```
model FriendRequest {
  createdAt        DateTime   @default(now()) @map(name: "created_at")
  to_user_id         Int       
  status           ReqStatus  @default(pending)
  from_user_id       Int        
  toUser           User       @relation("friend_requests_to", fields: [to_user_id], references: [id])
  fromUser         User       @relation("friend_requests_from", fields: [from_user_id], references: [id])

  @@id([from_user_id, to_user_id])
  @@map(name: "friend_requests")
}
```

and called `npx @prisma/cli@alpha generate` using this update call:
```
await prisma.friendRequest.update({
    where: {
        from_user_id_to_user_id: {from_user_id: 1, to_user_id: 2}
    }, 
    data: {status: "accepted"}
});
```

everything works fine

## Expected behavior

I would expect the status field being updated in the database

## Prisma information

Shared above

## Environment & setup

- OS: macOS 10.15.2
- Database: PostgreSQL
- Prisma version: 2.0.0-alpha.1174
- Node.js version: 13.12.0

**Comment by Sytten** [comment]
A good improvement to the `@@id` syntax would be able to name the damn thing IMO.

**Comment by pantharshit00** [comment]
@Sytten New feature request please 🙏. I like the idea though.

Sorry for the late reply here, this issue got buried in my notifications.

I can confirm this issue. I made a small reproduction here for internal use as we need to add some parts to your schema to make it complete: https://github.com/harshit-test-org/prisma-issue-2329

Version used:
```
Environment variables loaded from ./prisma/.env
@prisma/cli          : 2.0.0-alpha.1217
Current platform     : darwin
Query Engine         : query-engine 0b7f6a9f901fa5635e23a8d5f1dbe0ec273a3a11 (at /Users/harshit/.npm/_npx/90136/lib/node_modules/@prisma/cli/query-engine-darwin)
Migration Engine     : migration-engine-cli 0b7f6a9f901fa5635e23a8d5f1dbe0ec273a3a11 (at /Users/harshit/.npm/_npx/90136/lib/node_modules/@prisma/cli/migration-engine-darwin)
Introspection Engine : introspection-core 0b7f6a9f901fa5635e23a8d5f1dbe0ec273a3a11 (at /Users/harshit/.npm/_npx/90136/lib/node_modules/@prisma/cli/introspection-engine-darwin)
Format Binary        : prisma-fmt 0b7f6a9f901fa5635e23a8d5f1dbe0ec273a3a11 (at /Users/harshit/.npm/_npx/90136/lib/node_modules/@prisma/cli/prisma-fmt-darwin)

```

**Comment by dpetrick** [comment]
Fixed from 2.0.0-alpha.1230 onwards.
